xlsx/src/import/shared_strings.rs

```diff
@@ -1,39 +1,121 @@
-use std::io::Read;
+use std::io::{BufReader, Read};
 
+use quick_xml::events::Event;
 use roxmltree::Node;
 
 use crate::error::XlsxError;
 
+struct SSParser {
+    state: ParserState,
+    current_string: String,
+    strings: Vec<String>,
+}
+
+impl SSParser {
+    fn new() -> Self {
+        Self {
+            state: ParserState::OutsideSi,
+            current_string: String::new(),
+            strings: vec![],
+        }
+    }
+
+    fn process(&mut self, ev: Event) -> Result<(), XlsxError> {
+        self.state = match self.state {
+            ParserState::OutsideSi => match ev {
+                Event::Start(e) if e.local_name().into_inner() == b"si" => ParserState::InsideSi,
+                _ => ParserState::OutsideSi,
+            },
+            ParserState::InsideSi => match ev {
+                Event::Start(e) if e.local_name().into_inner() == b"t" => ParserState::T,
+                Event::End(e) if e.local_name().into_inner() == b"si" => {
+                    self.strings.push(self.current_string.clone());
+                    self.current_string.clear();
+                    ParserState::OutsideSi
+                }
+                _ => ParserState::InsideSi,
+            },
+            ParserState::T => match ev {
+                Event::Text(t) => {
+                    self.current_string
+                        .push_str(t.unescape().unwrap_or("".into()).as_ref());
+                    ParserState::T
+                }
+                Event::End(e) if e.local_name().into_inner() == b"t" => ParserState::InsideSi,
+                _ => ParserState::T,
+            },
+        };
+
+        Ok(())
+    }
+
+    fn strings(self) -> Result<Vec<String>, XlsxError> {
+        if self.state != ParserState::OutsideSi {
+            return Err(XlsxError::Xml("Corrupt XML structure".to_string()));
+        }
+        Ok(self.strings)
+    }
+}
+
+#[derive(Debug, PartialEq)]
+enum ParserState {
+    OutsideSi,
+    InsideSi,
+    T,
+}
+
```

Comment:
Pretty simple parser for the shared strings file. It isn't too big, but this gives us some memory savings and some time savings.